test/models/user_test.rb

```diff
@@ -320,6 +320,32 @@ def setup_user(operation)
     assert_includes record.errors.keys, :admin
   end
 
+  test "log of password change should be saved" do
+    user = FactoryGirl.create(:user)
+    user.password = nil
+    user.save
+    as_admin do
+      user.password = "newpassword"
+      user.valid?
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+    end
+  end
+
+  test "log of password change shouldnt be saved" do
```

Comment:
1. shouldn't, missing an apostrophe
1. There is no password change here - it isn't that the log of a password change shouldn't be saved, it's that there is no audit log entry created when there is no password change.